Sources/SwiftProtobuf/TextScanner.swift

```diff
@@ -264,14 +259,16 @@ private func decodeString(_ s: String) -> String? {
 ///
 /// TextScanner has no public members.
 ///
-public class TextScanner {
+public struct TextScanner {
```

Comment:
I made the TextScanner a struct mainly for consistency.  It seems to make very little performance difference whether you (a) copy the scanner into a sub-decoder and copy back or (b) refcount a class reference.  This is especially true for JSON, where the scanner state is literally just a pointer and a count.